fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -1434,6 +1434,192 @@ module.exports = function(config, DB) {
+            var sessionId3 = hex32()
+
+            var session1 = {
+              data: hex32(),
+              uid: uid1,
+              createdAt: Date.now(),
+              uaBrowser: 'foo',
+              uaBrowserVersion: 'bar',
+              uaOS: 'baz',
+              uaOSVersion: 'qux',
+              uaDeviceType: 'wibble',
+              mustVerify: false,
+              tokenVerificationId: hex16()
+            }
+
+            var session2 = {
+              data: hex32(),
+              uid: uid1,
+              createdAt: Date.now(),
+              uaBrowser: 'foo',
+              uaBrowserVersion: 'bar',
+              uaOS: 'baz',
+              uaOSVersion: 'qux',
+              uaDeviceType: 'wibble'
+              // no tokenVerificationId means verified
+            }
+
+            var session3 = {
+              data: hex32(),
+              uid: uid1,
+              createdAt: Date.now(),
+              uaBrowser: 'foo',
+              uaBrowserVersion: 'bar',
+              uaOS: 'baz',
+              uaOSVersion: 'qux',
+              uaDeviceType: 'wibble',
+              mustVerify: false,
+              tokenVerificationId: hex16()
+            }
+
+            return securityEvents({
+              setUp: function () {
+                return P.all([
+                  createSession(sessionId1, session1),
+                  createSession(sessionId2, session2),
+                  createSession(sessionId3, session3)
+                ])
+                .then(function () {
+                  return P.all([
+                    insert(uid1, addr1, evB, sessionId1),
+                    insert(uid1, addr1, evA, sessionId2),
+                    insert(uid1, addr2, evA, sessionId3),
+                    insert(uid2, addr1, evA, hex32())
+                  ])
+                })
+              },
+
+              testGetEvent: query(
+                uid1, addr1,
+                function (results) {
+                  t.equal(results.length, 2, 'two events for uid and addr')
+                  t.equal(results[0].name, evB, 'correct event name')
+                  t.equal(results[0].verified, false, 'first session is not verified yet')
+                  t.ok(results[0].createdAt < Date.now(), 'createdAt is set')
+                  t.equal(results[1].name, evA, 'correct event name')
+                  t.equal(results[1].verified, true, 'second session is verified')
+                  t.ok(results[1].createdAt < Date.now(), 'createdAt is set')
+                }
+              ),
+
+              testGetEventAfterSessionVerified: function () {
+                return verifySession(session1.tokenVerificationId, uid1)
+                  .then(query(uid1, addr2, function (results) {
+                    t.equal(results.length, 2, 'two events for uid and addr')
+                    t.equal(results[0].verified, true, 'first session verified')
+                    t.equal(results[1].verified, true, 'second session verified')
```

Comment:
It's probably worth adding a test to ensure that other events for this user were *not* verified, to guard against e.g. a bug in the `verifyToken_2` query that accidentally selects all rows for the user.